pkgs/sdk/server/src/Configuration.cs

```diff
@@ -187,7 +187,7 @@ internal Configuration(ConfigurationBuilder builder)
             Events = builder._events;
             Http = builder._http;
             Logging = builder._logging;
-            Offline = builder._offline;
+            Offline = builder._sdkKey == null || builder._offline;
```

Comment:
Do we want to force offline mode here? It seems unnecessary to attempt connecting if we know the key isn't valid.